### DISCUSSION TITLE
Force NER model to output only one entity

### DISCUSSION THREAD

**Discussion by lanchiang** [explosion/spaCy#12223] [Help: Best practices]
Hello,

I have trained a NER model to find the keyword (entity) in a term that may have several words. In my use case, one and only one entity should be found. But I notice that spaCy's NER model sometimes output more than one entity, and sometimes no entities at all. **I wonder if there's a way to achieve what I want with spaCy?**

P.S., I have thought of modelling the problem as text classification where each keyword is a class. But the problem is the classifier won't be able to recognise classes not in the training set.

I have tried to look for similar discussions here but cannot seemingly find them.
Any help would be greatly appreciated.

Thank you!

**Comment by rmitsch** [comment]
Hey @lanchiang, can you elaborate on what exactly you're trying to achieve? If you have only entity you want to identify, it might be more practical to use rule-based matching. It would help us to know more about your use-case though before recommending next steps.

**Reply by lanchiang** [reply]
Hi @rmitsch, thanks for the reply.
Say we want to detect the position of surname in a person's full name. Assume that each name only has one surname and a full name may have several words. Some people place their surname left-most, some people right-most. Some people might use one in the middle of the name as their surname.

In this case, classification would be hard because surnames are diverse and cannot be all covered in the training data. I am thinking of the rule-based matching you said. But I guess it would also be hard because of the same reason?

**Reply by rmitsch** [reply]
Yes, it'd be rather difficult. I don't expect that a model would be able to learn differentiating between surnames and first names well - it's a somewhat ill-defined problem, as surnames and first names can be identical. Even in cases where they aren't, a learner would likely need additional context to distinguish them reliably.

An alternative might be collecting common surnames and fuzzily matching to those, but that's heavy-handed and error-prone. I recommend taking a step back and rethinking how you approach this problem. Maybe there's a way to obtain additional context, or perhaps you can solve this differently/without NLP.
